internal/statemachine/classic_states.go

```diff
@@ -1313,30 +1312,20 @@ func (stateMachine *StateMachine) makeQcow2Img() error {
 // updateBootloader determines the bootloader for each volume
 // and runs the correct helper function to update the bootloader
 func (stateMachine *StateMachine) updateBootloader() error {
-	// determine which partition number is the rootfs and which volume it is in
-	// TODO should this be stored in the struct earlier on?
-	rootfsPartNum := -1
-	for _, volumeName := range stateMachine.VolumeOrder {
-		volume := stateMachine.GadgetInfo.Volumes[volumeName]
-		for structureNumber, structure := range volume.Structure {
```

Comment:
So the point here is that this is wrong because it needs to skip non partition elements of the structure? And rather than fixing that directly, instead look for the rootfs in an earlier iteration over the structure that gets this detail correct? If so, that sounds good to me!